src/controllers/controllermappinginfo.cpp

```diff
@@ -19,102 +30,156 @@ MappingInfo::MappingInfo(const QString& mapping_path)
     QFileInfo fileInfo(mapping_path);
     m_path = fileInfo.absoluteFilePath();
     m_dirPath = fileInfo.dir().absolutePath();
-    m_name = "";
-    m_author = "";
-    m_description = "";
-    m_forumlink = "";
-    m_wikilink = "";
-
-    QDomElement root = XmlParse::openXMLFile(m_path, "controller");
-    if (root.isNull()) {
-        qWarning() << "ERROR parsing" << m_path;
-        return;
-    }
-    QDomElement info = root.firstChildElement("info");
-    if (info.isNull()) {
-        qDebug() << "MISSING <info> ELEMENT: " << m_path;
+
+    QFile file(m_path);
+    if (!file.open(QIODevice::ReadOnly | QIODevice::Text)) {
+        qCWarning(kLogger) << "ERROR opening" << m_path;
         return;
     }
 
-    m_valid = true;
+    QXmlStreamReader xml(&file);
+    bool inInfo = false;
+    bool inInfoDevices = false;
+    int xmlHierachyDepth = 0;
 
-    QDomElement dom_name = info.firstChildElement("name");
-    if (!dom_name.isNull()) {
-        m_name = dom_name.text();
-    } else {
-        m_name = fileInfo.baseName();
-    }
+    while (!xml.atEnd()) {
+        QXmlStreamReader::TokenType token = xml.readNext();
+        if (token == QXmlStreamReader::StartElement) {
+            ++xmlHierachyDepth;
+            const QString xmlElementName = xml.name().toString();
```

Comment:
toString is a deep copy from a QStringView

https://github.com/qt/qtbase/blob/6d43724a30a636aa1bb719e871d23b9d0e0bc127/src/corelib/serialization/qxmlstream_p.h#L61

I think this should work: 

```suggestion
            const QStringView xmlElementName = xml.name();
```